src/component/form/AttribElement.tsx

```diff
@@ -1,37 +1,56 @@
+import * as isEqual from 'fast-deep-equal';
 import * as React from 'react';
 import { connect } from 'react-redux';
 import { v4 as generateUUID } from 'uuid';
-
 import { AttributeType, ResultType } from '../../flowTypes';
-import { AppState, SearchResult, UpdateContactFields, updateContactFields } from '../../store';
+import { AppState, SearchResult } from '../../store';
 import { getSelectClass, propertyExists } from '../../utils';
 import SelectSearch from '../SelectSearch';
 import FormElement, { FormElementProps } from './FormElement';
 
 interface AttribElementPassedProps extends FormElementProps {
     initial: SearchResult;
-    endpoint?: string;
+    endpoint: string;
     add?: boolean;
     placeholder?: string;
     searchPromptText?: string;
+    helpText?: string;
 }
 
 interface AttribElementStoreProps {
     contactFields: SearchResult[];
-    updateContactFields: UpdateContactFields;
 }
 
 export type AttribElementProps = AttribElementPassedProps & AttribElementStoreProps;
 
-interface AttribState {
+interface AttribElementState {
     attribute: SearchResult;
     errors: string[];
 }
 
 export const PLACEHOLDER = 'Enter the name of an existing attribute or create a new one';
 export const NOT_FOUND = 'Invalid attribute name';
+export const VALID_FIELD = /^[a-z0-9-][a-z0-9- ]*$/;
+export const CREATE_PROMPT = 'New attribute: ';
+
+export const attribExists = (newOptName: string, options: SearchResult[]) => {
+    const loweredNTrimmed = newOptName.toLowerCase().trim();
+    if (options.length) {
```

Comment:
for things like this, what about instead doing 
```javascript
options.find((searchResult: SearchResult)=> ...
```
...would remove a branch in coverage too

Reply:
much cleaner, addressed here https://github.com/nyaruka/floweditor/pull/252/commits/31b503b1cca54427639dcf70ad6cda885b3b0775